### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by babiracka**
Searching by description, title, country and users.

### CODE REVIEW

**Comment by rishijain** [comment]
👍 Thanks. Will look at it and merge it.

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -27,9 +27,26 @@ def update
       render 'edit'
     end
   end
-
+# # search
```
Comment (MacDada) [inline app/controllers/posts_controller.rb:30]: @babiracka Don't commit commented out code

File: app/models/user.rb [hunk 2]
```diff
@@ -63,4 +64,10 @@ def self.from_omniauth(auth)
 
     return user
   end
+
+  #rebuild
```
Comment (MacDada) [inline app/models/user.rb:68]: @babiracka the method has `rebuild` in its name – I don't think the comment is necessary in this case

File: app/models/post.rb [hunk 3]
```diff
@@ -15,7 +28,7 @@ class Post < ActiveRecord::Base
 
   scope :all_except, -> (post_id) { where.not(id: post_id) }
   self.per_page = 3
-
+
```
Comment (MacDada) [inline app/models/post.rb:31]: @babiracka superfluous whitespace

File: app/models/post.rb [hunk 4]
```diff
@@ -1,11 +1,24 @@
 class Post < ActiveRecord::Base
 
   include AASM
-
+  include PgSearch
   belongs_to :user
   has_many :comments
   has_many :likes
-
+  #search:
```
Comment (MacDada) [inline app/models/post.rb:8]: @babiracka same as [here](https://github.com/rishijain/oneplaceaday/pull/55/files#r36463706) – a comment doesn't add any value when it duplicates words from the method it describes

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -27,9 +27,26 @@ def update
...
+#     Post.search params[:query]
+#   end
+  
   def index
-    @posts = Post.published.order('views_count DESC , created_at DESC')
+    @posts = Post.all
+    if params[:query].present?
+      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
+    else
+      @posts = Post.published.order('views_count DESC , created_at DESC')
```
Comment (MacDada) [inline app/controllers/posts_controller.rb:48]: @babiracka There is a small duplication: `.order('views_count DESC , created_at DESC')`.

Imagine you add more conditions. Or more methods need to be called on the query. Are you going to copy–paste it to every case?

You could put that call outside the if statements instead :)

Comment (rishijain) [inline app/controllers/posts_controller.rb:48]: Agreed with @MacDada . Also do we need to do collect all the posts first before getting to if/else block?  @babiracka

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (rishijain) [inline app/controllers/posts_controller.rb:30]: Yes @babiracka . And if you notice the first commit I made after merging your PR was removing commented code and extra white spaces. Here (https://github.com/rishijain/oneplaceaday/commit/05b180bdbda15e8d452f9a9366974e5f361dc5a1)

**Pull request merged**